### DISCUSSION TITLE
Mountain duck Upload file 423 locked error

### DISCUSSION THREAD

**Discussion by meiranzheng** [iterate-ch/cyberduck#16586] [Q&A]
## Background

I have a 200 Mbps bandwidth connection, and I use Alist to mount China Mobile and Cailian Cloud (彩云网盘) to provide WebDAV services. Downloading files works fine, but I encounter an issue when uploading files larger than 2GB — the process consistently results in a 423 Locked Error.

I have tested Mountain Duck versions 4.15 through 4.16.3, and even the 4.17 beta, but the issue persists. Occasionally, I've had a file upload of about 4GB succeed, but this is rare.

### Troubleshooting

After numerous attempts, including monitoring Alist server logs, local network traffic, and other factors, it seems that the issue occurs after the upload completes, during the waiting period for the cloud storage service to respond. While this may sound odd, it’s actually the case.

For instance, uploading a 7GB file takes around 3 to 4 minutes in my network environment, but after the upload is complete, I have to wait another 3 to 4 minutes to receive a completion response from the cloud storage service.

I am unsure if the response delay is related to the file size or upload traffic, but it could be due to the cloud storage service performing additional file security or compliance checks.

### Current Limitation

It seems that Alist does not intend to address this issue, since uploading through other WebDAV clients (with a longer wait time) does allow the upload to complete. Additionally, Alist focuses primarily on download services, with file uploads being a secondary feature.

However, for my use case with China Mobile and Cailian Cloud, Alist appears to be the only viable option for mounting the cloud storage.

### Request

I would like to request that Mountain Duck provide more flexible custom connection timeout settings, allowing users to set longer timeouts, particularly for scenarios involving large file uploads to cloud storage. Currently, Mountain Duck’s connection timeout limit seems to be capped at 60 seconds, which is clearly insufficient for large file uploads.

### Suggestion

I suggest Mountain Duck increase the adjustable timeout range, with a minimum of 5 minutes. Relaxing the timeout limit could help users better handle response delays from cloud storage services, especially when uploading large files.

I understand that opening up the timeout limit completely could lead to some users setting excessively long timeouts, but I believe this could be mitigated in the following ways:
- Provide a warning or tooltip when users modify the timeout setting, suggesting reasonable timeout values.
- After multiple upload failures, prompt users to adjust their timeout settings.

### Conclusion

I believe that by providing more flexible timeout options, Mountain Duck can improve compatibility with certain cloud storage services, ultimately offering a better user experience for large file uploads.

Thank you for the hard work of the development team and the community. I wish everyone a wonderful day!

<img width="1600" alt="image" src="https://github.com/user-attachments/assets/dfad7d14-f8d5-44b2-b3a4-b02f511562a2">

**Comment by meiranzheng** [comment]
<img width="690" alt="image" src="https://github.com/user-attachments/assets/89ad5bd4-618f-48d2-a70e-68e64e9e8fab">
I have tried various boundary file sizes, including generating files of various sizes with the dd command.   If the proposal is not adopted, I hope you can guide me on how to modify the 60s limit of the timeout connection in the existing version.thanks

**Comment by meiranzheng** [comment]
I tried to modify the local configuration file, but it seems that the mountain dock software has a logical limit on the number of seconds. Even if it is changed to 300s, only 60s is displayed and used. So can anyone help me solve this problem? If possible, I hope this proposal can be supported in the preview version.
<img width="645" alt="image" src="https://github.com/user-attachments/assets/673c5fe2-891a-4b8c-899b-e23de710c0f6">

**Comment by dkocher** [comment]
In [^1]. I can confirm regardless of the setting the maximum timeout is 60s with #12979.

[^1]: #16587.
